### PR TITLE
Issue #566:Customize HTTP status codes for Prometheus exporter

### PR DESCRIPTION
**Pull request by epaulsen**
**What this PR does / why we need it**:

This is needed in order to have prometheus read metrics when health check status is failing.

**Which issue(s) this PR fixes**: #566 

Please reference the issue this PR will close: #566 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**: No

Please make sure you've completed the relevant tasks for this PR, out of the following list:

- [X] Code compiles correctly
- [X] Created/updated tests
- [X] Unit tests passing
- [X] End-to-end tests passing
- [X] Extended the documentation
- [ ] Provided sample for the feature

### CODE REVIEW

File: test/FunctionalTests/HealthChecks.Prometheus.Metrics/PrometheusResponseWriterTests.cs [hunk 1]
```diff
@@ -67,5 +67,28 @@ public async Task be_unhealthy_and_return_503_when_health_checks_are()
...
+            var sut = new TestServer(new WebHostBuilder()
+                .UseStartup<DefaultStartup>()
+                .ConfigureServices(services =>
+                {
+                    services.AddHealthChecks()
+                        .AddCheck("fake", check => HealthCheckResult.Unhealthy());
+                })
+                .Configure(app =>
+                {
+                    app.UseHealthChecksPrometheusExporter("/health",options=>options.ResultStatusCodes[HealthStatus.Unhealthy] = (int)HttpStatusCode.OK);
```
Comment (sungam3r) [inline test/FunctionalTests/HealthChecks.Prometheus.Metrics/PrometheusResponseWriterTests.cs:83]:
```suggestion
                    app.UseHealthChecksPrometheusExporter("/health",options => options.ResultStatusCodes[HealthStatus.Unhealthy] = (int)HttpStatusCode.OK);
```

File: src/HealthChecks.Prometheus.Metrics/Extensions/ApplicationBuilderExtensions.cs [hunk 2]
```diff
@@ -1,17 +1,20 @@
...
+using HealthChecks.Prometheus.Metrics;
 using Microsoft.AspNetCore.Diagnostics.HealthChecks;
 using Microsoft.AspNetCore.Http;
 
 namespace Microsoft.AspNetCore.Builder
 {
     public static class PrometheusHealthCheckMiddleware 
     {
-        public static IApplicationBuilder UseHealthChecksPrometheusExporter(this IApplicationBuilder applicationBuilder, PathString endpoint) 
+        public static IApplicationBuilder UseHealthChecksPrometheusExporter(this IApplicationBuilder applicationBuilder, PathString endpoint, Action<HealthCheckOptions> configure = null)
```
Comment (sungam3r) [inline src/HealthChecks.Prometheus.Metrics/Extensions/ApplicationBuilderExtensions.cs:10]: This change is still binary breaking.

Comment (epaulsen) [reply src/HealthChecks.Prometheus.Metrics/Extensions/ApplicationBuilderExtensions.cs:10]: Ok.  Would it be better if I added the `Action<HealthCheckOptions>` parameter as an overloaded method instead?

Comment (sungam3r) [reply src/HealthChecks.Prometheus.Metrics/Extensions/ApplicationBuilderExtensions.cs:10]: This is usually done when you want to guarantee 100% backward compatibility.

Comment (epaulsen) [reply src/HealthChecks.Prometheus.Metrics/Extensions/ApplicationBuilderExtensions.cs]: Moved options to an overloaded method.

**Comment by unaizorrilla** [comment]
Merged!

A new release build is on appveyor when finished a new package version 3.1.4 will be available on nuget.

Thanks @epaulsen  for this contribution!!

**Pull request merged**